openedx/core/djangoapps/config_model_utils/models.py

```diff
@@ -0,0 +1,125 @@
+    site = models.ForeignKey(Site, on_delete=models.CASCADE, null=True)
+    org = models.CharField(max_length=255, db_index=True, null=True)
+    course = models.ForeignKey(
+        CourseOverview,
+        on_delete=models.DO_NOTHING,
+        null=True,
+    )
+
+    @classmethod
+    def attribute_tuple(cls):
+        "Returns a namedtuple with all attributes that can be overridden on this config model."
+        if hasattr(cls, '_attribute_tuple'):
+            return cls._attribute_tuple
+
+        cls._attribute_tuple = namedtuple(
+            '{}Values'.format(cls.__name__),
+            cls.STACKABLE_FIELDS
+        )
+        return cls._attribute_tuple
+
+    @classmethod
+    def current(cls, site=None, org=None, course=None):  # disable: pylint=arguments-differ
+        """
+        Return the current overridden configuration at the specified level.
+
+        Only one level may be specified at a time. Specifying multiple levels
+        will result in a ValueError.
+
+        Arguments:
+            site: The Site to check current values for
+            org: The org to check current values for
+            course: The course to check current values for
+
+        Returns:
+            An instance of :class:`cls.attribute_tuple()` with the overridden values
+            specified down to the level of the supplied argument (or global values if
+            no arguments are supplied).
+        """
+        if len([arg for arg in [site, org, course] if arg is not None]) > 1:
+            raise ValueError("Only one of site, org, and course can be specified")
+
+        if org is None and course is not None:
+            org = cls._org_from_course(course)
+
+        if site is None and org is not None:
+            site = cls._site_from_org(org)
+
+        stackable_fields = [cls._meta.get_field(field_name) for field_name in cls.STACKABLE_FIELDS]
+        field_defaults = {
+            field.name: field.get_default()
+            for field in stackable_fields
+        }
+
+        values = field_defaults.copy()
+
+        global_current = super(StackedConfigurationModel, cls).current(None, None, None)
+        for field in stackable_fields:
+            values[field.name] = field.value_from_object(global_current)
+
+        if site is not None:
+            site_current = super(StackedConfigurationModel, cls).current(site, None, None)
+            for field in stackable_fields:
+                value = field.value_from_object(site_current)
+                if value != field_defaults[field.name]:
+                    values[field.name] = value
+
+        if org is not None:
+            org_current = super(StackedConfigurationModel, cls).current(None, org, None)
+            for field in stackable_fields:
+                value = field.value_from_object(org_current)
+                if value != field_defaults[field.name]:
+                    values[field.name] = value
+
+        if course is not None:
```

Comment:
it seems this code is repeated 3 times except for different arguments/variable names.  
maybe it would be clearer to just pull it out into a separate function called 3 times from this function?

Reply:
Maybe. I'll look and see what I can do about abstracting it a bit (maybe a fn, maybe a loop)